Content.Shared/Stunnable/SharedStunSystem.Knockdown.cs

```diff
@@ -0,0 +1,469 @@
+        SubscribeLocalEvent<KnockedDownComponent, BuckleAttemptEvent>(OnBuckleAttempt);
+        SubscribeLocalEvent<KnockedDownComponent, StandAttemptEvent>(OnStandAttempt);
+
+        // Updating movement a friction
+        SubscribeLocalEvent<KnockedDownComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshKnockedSpeed);
+        SubscribeLocalEvent<KnockedDownComponent, RefreshFrictionModifiersEvent>(OnRefreshFriction);
+        SubscribeLocalEvent<KnockedDownComponent, TileFrictionEvent>(OnKnockedTileFriction);
+        SubscribeLocalEvent<KnockedDownComponent, DidEquipHandEvent>(OnHandEquipped);
+        SubscribeLocalEvent<KnockedDownComponent, DidUnequipHandEvent>(OnHandUnequipped);
+
+        // DoAfter event subscriptions
+        SubscribeLocalEvent<KnockedDownComponent, TryStandDoAfterEvent>(OnStandDoAfter);
+
+        // Knockdown Extenders
+        SubscribeLocalEvent<KnockedDownComponent, DamageChangedEvent>(OnDamaged);
+
+        // Handling Alternative Inputs
+        SubscribeAllEvent<ForceStandUpEvent>(OnForceStandup);
+        SubscribeLocalEvent<KnockedDownComponent, KnockedDownAlertEvent>(OnKnockedDownAlert);
+
+        CommandBinds.Builder
+            .Bind(ContentKeyFunctions.ToggleKnockdown, InputCmdHandler.FromDelegate(HandleToggleKnockdown, handle: false))
+            .Register<SharedStunSystem>();
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<KnockedDownComponent>();
+
+        while (query.MoveNext(out var uid, out var knockedDown))
+        {
+            if (!knockedDown.AutoStand || knockedDown.DoAfterId.HasValue || knockedDown.NextUpdate > GameTiming.CurTime)
+                continue;
+
+            TryStanding(uid, out knockedDown.DoAfterId);
+            DirtyField(uid, knockedDown, nameof(KnockedDownComponent.DoAfterId));
+        }
+    }
+
+    private void OnRejuvenate(Entity<KnockedDownComponent> entity, ref RejuvenateEvent args)
+    {
+        SetKnockdownTime(entity, GameTiming.CurTime);
+
+        if (entity.Comp.AutoStand)
+            RemComp<KnockedDownComponent>(entity);
+    }
+
+    #region Startup and Shutdown
+
+    private void OnKnockInit(Entity<KnockedDownComponent> entity, ref ComponentInit args)
+    {
+        _standingState.Down(entity, true, entity.Comp.AutoStand);
+        RefreshKnockedMovement(entity);
+    }
+
+    private void OnKnockShutdown(Entity<KnockedDownComponent> entity, ref ComponentShutdown args)
+    {
+        // This is jank but if we don't do this it'll still use the knockedDownComponent modifiers for friction because it hasn't been deleted quite yet.
+        entity.Comp.FrictionModifier = 1f;
+        entity.Comp.SpeedModifier = 1f;
+
+        _standingState.Stand(entity);
+        Alerts.ClearAlert(entity, KnockdownAlert);
+    }
+
+    #endregion
+
+    #region API
+
+    public void ToggleAutoStand(Entity<KnockedDownComponent?> entity, bool autoStand = false)
```

Comment:
```suggestion
    public void SetAutoStand(Entity<KnockedDownComponent?> entity, bool autoStand = false)
```
this is not toggling it
also add documenation to all public methods here